tests/unit/test_models/test_full_battery_models/test_lead_acid/test_full.py

```diff
@@ -1,55 +1,56 @@
-
-        options = {"dimensionality": 1, "convection": "full transverse"}
+    @pytest.mark.parametrize(
+        "options",
+        [
+            {},
+            {"convection": "uniform transverse"},
+            {"dimensionality": 1, "convection": "full transverse"},
+        ],
+        ids=["well_posed", "with_convention", "with_convention_1plus1d"],
+    )
+    def test_well_posed(self, options):
         model = pybamm.lead_acid.Full(options)
         model.check_well_posedness()
 
 
 class TestLeadAcidFullSurfaceForm:
-    def test_well_posed_differential(self):
-        options = {"surface form": "differential"}
-        model = pybamm.lead_acid.Full(options)
-        model.check_well_posedness()
-
-    def test_well_posed_differential_1plus1d(self):
-        options = {"surface form": "differential", "dimensionality": 1}
-        model = pybamm.lead_acid.Full(options)
-        model.check_well_posedness()
-
-    def test_well_posed_algebraic(self):
-        options = {"surface form": "algebraic"}
+    @pytest.mark.parametrize(
+        "options",
+        [
+            {"surface form": "differential"},
+            {"surface form": "differential", "dimensionality": 1},
+            {"surface form": "algebraic"},
+        ],
+        ids=["differential", "differential_1plus1d", "algebraic"],
+    )
+    def test_well_posed(self, options):
         model = pybamm.lead_acid.Full(options)
         model.check_well_posedness()
 
 
 class TestLeadAcidFullSideReactions:
-    def test_well_posed(self):
-        options = {"hydrolysis": "true"}
-        model = pybamm.lead_acid.Full(options)
-        model.check_well_posedness()
-
-    def test_well_posed_surface_form_differential(self):
-        options = {"hydrolysis": "true", "surface form": "differential"}
-        model = pybamm.lead_acid.Full(options)
-        model.check_well_posedness()
-        assert isinstance(model.default_solver, pybamm.CasadiSolver)
-
-    def test_well_posed_surface_form_algebraic(self):
-        options = {"hydrolysis": "true", "surface form": "algebraic"}
-        model = pybamm.lead_acid.Full(options)
-        model.check_well_posedness()
-        assert isinstance(model.default_solver, pybamm.CasadiSolver)
+    @pytest.mark.parametrize(
+        "options, expected_solver",
+        [
+            ({"hydrolysis": "true"}, None),
+            (
+                {"hydrolysis": "true", "surface form": "differential"},
+                pybamm.CasadiSolver,
+            ),
+            ({"hydrolysis": "true", "surface form": "algebraic"}, pybamm.CasadiSolver),
+        ],
+        ids=[
+            "well_posed",
+            "surface_form_differential",
+            "surface_form_algebraic",
+        ],
+    )
+    def test_well_posed(self, options, expected_solver):
+        model = pybamm.lead_acid.Full(options)
+        model.check_well_posedness()
+        if expected_solver is not None:
+            assert isinstance(model.default_solver, expected_solver)
```

Comment:
I think this case is a bit different and difficult to follow, as it branches out within a test (for no good reason). Could we move the case where `expected_solver` is `None` back to its own test?